File: ueueo-framework/ueueo-authorization/src/main/java/com/ueueo/authorization/permissions/ClientPermissionValueProvider.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Volo.Abp.MultiTenancy;
using System.Linq;
using Volo.Abp.Security.Claims;

namespace Volo.Abp.Authorization.Permissions;

public class ClientPermissionValueProvider : PermissionValueProvider
{
    public const String ProviderName = "C";

    @Override
    public String Name => ProviderName;

    protected ICurrentTenant CurrentTenant;//  { get; }

    public ClientPermissionValueProvider(IPermissionStore permissionStore, ICurrentTenant currentTenant)
        : base(permissionStore)
    {
        CurrentTenant = currentTenant;
    }

    @Override
    public Task<PermissionGrantResult> CheckAsync(PermissionValueCheckContext context)
    {
        var clientId = context.Principal?.FindFirst(AbpClaimTypes.ClientId)?.Value;

        if (clientId == null)
        {
            return PermissionGrantResult.Undefined;
        }

        using (CurrentTenant.Change(null))
        {
            return PermissionStore.IsGrantedAsync(context.Permission.Name, Name, clientId)
                ? PermissionGrantResult.Granted
                : PermissionGrantResult.Undefined;
        }
    }

    @Override
    public Task<MultiplePermissionGrantResult> CheckAsync(PermissionValuesCheckContext context)
    {
        var permissionNames = context.Permissions.Select(x => x.Name).Distinct().ToArray();
        Check.NotNullOrEmpty(permissionNames, nameof(permissionNames));

        var clientId = context.Principal?.FindFirst(AbpClaimTypes.ClientId)?.Value;
        if (clientId == null)
        {
            return new MultiplePermissionGrantResult(permissionNames); ;
        }

        using (CurrentTenant.Change(null))
        {
            return PermissionStore.IsGrantedAsync(permissionNames, Name, clientId);
        }
    }
}

File: ueueo-framework/ueueo-authorization/src/main/java/com/ueueo/authorization/permissions/ClientPermissionValueProvider.java
package com.ueueo.authorization.permissions;

import com.ueueo.ID;
import com.ueueo.multitenancy.ICurrentTenant;
import org.apache.commons.collections4.CollectionUtils;
import org.springframework.util.Assert;

import java.util.List;
import java.util.stream.Collectors;

public class ClientPermissionValueProvider extends PermissionValueProvider {

    public final String ProviderName = "C";

    @Override
    public String getName() {
        return ProviderName;
    }

    protected ICurrentTenant currentTenant;

    public ClientPermissionValueProvider(IPermissionStore permissionStore, ICurrentTenant currentTenant) {
        super(permissionStore);
        this.currentTenant = currentTenant;
    }

    @Override
    public PermissionGrantResult check(PermissionValueCheckContext context) {
        ID clientId = context.getPrincipal().findClientId();

        if (clientId == null) {
            return PermissionGrantResult.Undefined;
        }

        return permissionStore.isGranted(context.getPermission().getName(), getName(), clientId.getStringValue())
                ? PermissionGrantResult.Granted
                : PermissionGrantResult.Undefined;
    }

    @Override
    public MultiplePermissionGrantResult check(PermissionValuesCheckContext context) {
        List<String> permissionNames = context.getPermissions().stream().map(PermissionDefinition::getName).distinct().collect(Collectors.toList());
        Assert.isTrue(CollectionUtils.isNotEmpty(permissionNames), "PermissionNames must not empty!");

        ID clientId = context.getPrincipal().findClientId();
        if (clientId == null) {
            return new MultiplePermissionGrantResult(permissionNames);
        }

        return permissionStore.isGranted(permissionNames, getName(), clientId.getStringValue());
    }

    public ICurrentTenant getCurrentTenant() {
        return currentTenant;
    }
}
